module-2/backend/week_one.md

```diff
@@ -4,31 +4,77 @@ Fork this respository. Answer the questions to the best of your ability. Try to
-9. What's the purpose of ERB?
-10. Why do I need a development AND test database?
-11. What's responsive design?
-12. What is CRUD and why is it important?
-13. What does HTTP stand for? 
-14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
-15. What's an ORM?
-16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
-17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-18. What's a migration? 
-19. When you create a migration, does it automatically modify your database?
-20. How does a model relate to a database?
-21. What's the difference between agile workflow and waterfall method?
-22. What is the difference between `#new` and `#create`?
+Inside index.erb you would need to have a line that contains "<%= @horses.count%>"
+
+**8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?**
+You would need a code that contains "<%= @horses.name %>
+
+
+**9. What's the purpose of ERB?**
+An Embeded Ruby file process the ruby code behind the scenes and then renders it to the browser.
+
+**10. Why do I need a development AND test database?**
+To ensure that the one environment doesn't change the other as a test database is typically updated and modified during tests. 
+
+**11. What's responsive design?**
+It's a design  where one single code/design will be display adaptabily in the desktop, tables, mobile phones etc. 
+
+**12. What is CRUD and why is it important?**
+Stands for Create, Read, Update and Delete and are the main principles of database management. It makes a database easier to modify and it also helps facilitate testing.
+
+**13. What does HTTP stand for?**
+HyperText Transfer Protocol
+
+**14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?**
+By using <%= something %> and <% something %>. The first one allows ruby to be processed behind the scenes and render it's result to the view and the second one will only execute the code without displaying the result.
+
+**15. What's an ORM?**
+Stands for Object-Relational Mapping and it allows converting of data between incompatible systems.
+
+**16. What's the most commonly used ORM in ruby (Sinatra & Rails)?**
+ActiveRecord
+
+**17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.**
+
+**18. What's a migration?**
+It's a table alteration command and records that allows you to manage your table and database. 
+
+**19. When you create a migration, does it automatically modify your database?**
+Yes.

```

Comment:
No! You still have to run `rake db:migrate` to get the migration to actually change the database.